src/components/mols/TrialConversionModal/index.tsx

```diff
@@ -11,15 +10,13 @@ import { useSession } from "next-auth/react";
 import usersService from "@/service/users.service";
 import { SUBSCRIPTION_STATUS } from "@/constants/payment";
 
-interface IPremiumConversionModal {
-  isPremiumExpired: boolean;
-  setIsPremiumExpired: Dispatch<SetStateAction<boolean>>;
+interface ITrialConversionModal {
+  setShowModal: Dispatch<SetStateAction<boolean>>;
   userId: string;
 }
 
-const PremiumConversionModal: React.FC<IPremiumConversionModal> = ({
-  isPremiumExpired,
-  setIsPremiumExpired,
+const TrialConversionModal: React.FC<ITrialConversionModal> = ({
+  setShowModal,
```

Comment:
this is a bad patterner -  "prop drilling"

Reply:
create a function named "handleModal" and pass this rather than useState

Reply:
done